routing/route.cpp

```diff
@@ -354,6 +354,63 @@ void Route::Update()
   m_currentTime = 0.0;
 }
 
+void Route::AppendRoute(Route const & route)
+{
+  if (!route.IsValid())
+    return;
+
+  if (m_poly.GetPolyline().GetSize() != 0)
+  {
+    ASSERT(!m_turns.empty(), ());
+    ASSERT(!m_times.empty(), ());
+
+    // Remove road end point and turn instruction.
+    ASSERT_LESS(MercatorBounds::DistanceOnEarth(m_poly.End().m_pt, route.m_poly.Begin().m_pt),
+                2 /* meters */, ());
+    m_poly.PopBack();
+    CHECK(!m_turns.empty(), ());
+    ASSERT_EQUAL(m_turns.back().m_turn, turns::TurnDirection::ReachedYourDestination, ());
+    m_turns.pop_back();
+    CHECK(!m_times.empty(), ());
+    m_times.pop_back();
+    // Streets might not point to the last point of the path.
```

Comment:
Am I right that m_streets is sorted by the first element of the pair? If so, could you please add a CHECK or ASSERT instead of a comment? I.e., to assert that m_street does not have an item for the evicted last m_poly item.